analyses/new-feature-template.qmd

```diff
@@ -0,0 +1,1549 @@
+
+
+```{r shap_processing}
+# Combine data
+shap_df_filtered_long <- shap %>%
+  inner_join(
+    assessment_data %>%
+      select(meta_pin, meta_card_num, meta_township_code, meta_nbhd_code) %>%
+      rename(township_code = meta_township_code, neighborhood_code = meta_nbhd_code),
+    by = c("meta_pin", "meta_card_num")
+  ) %>%
+  select(township_code, all_of(shap_predictors)) %>%
+  pivot_longer(
+    cols = all_of(shap_predictors),
+    names_to = "feature",
+    values_to = "shap"
+  )
+
+# Calculate median values for the overall data
+overall_medians <- shap_df_filtered_long %>%
+  group_by(feature) %>%
+  summarize(
+    median_abs_shap = median(abs(shap), na.rm = TRUE),
+    median_shap = median(shap, na.rm = TRUE)
+  ) %>%
+  ungroup()
+
+# Rank features based on median absolute SHAP values
+ranked_overall_medians <- overall_medians %>%
+  arrange(desc(median_abs_shap)) %>%
+  mutate(rank_abs = row_number())
+
+# Function to create summary tables
+create_summary_table <- function(data, added_feature) {
+  added_feature_data <- data %>%
+    filter(feature == added_feature)
+
+  if (nrow(added_feature_data) == 0) {
+    return(NULL)
+  }
+
+  non_added_feature_data <- data %>%
+    filter(feature != added_feature)
+
+  combined_data <- bind_rows(added_feature_data, non_added_feature_data)
+
+  kable_chart <- combined_data %>%
+    select(feature, median_abs_shap, median_shap, rank_abs) %>%
+    mutate(across(where(is.numeric), round, 2))
+
+  datatable_chart <- datatable(kable_chart,
+    options = list(
+      scrollY = "300px",
+      scrollX = TRUE,
+      paging = FALSE,
+      searching = TRUE
+    ),
+    rownames = FALSE
+  )
+
+  return(datatable_chart)
+}
+
+# Generate the overall summary table
+added_feature <- params$added_feature
+overall_summary_table <- create_summary_table(ranked_overall_medians, added_feature)
+
+# Calculate median values for each township and rank features
+ranked_shap_df <- shap_df_filtered_long %>%
+  group_by(township_code, feature) %>%
+  mutate(
+    median_abs_shap = median(abs(shap), na.rm = TRUE),
+    median_shap = median(shap, na.rm = TRUE)
+  ) %>%
+  ungroup() %>%
+  distinct(township_code, feature, .keep_all = TRUE) %>%
+  group_by(township_code) %>%
+  arrange(desc(median_abs_shap), .by_group = TRUE) %>%
+  mutate(rank_abs = row_number()) %>%
+  ungroup() %>%
+  inner_join(ccao::town_dict, by = c("township_code" = "township_code"))
+
+# Generate the township-specific tables
+township_names <- unique(ranked_shap_df$township_name)
+shap_tables <- lapply(township_names, function(name) {
+  township_data <- ranked_shap_df %>%
+    filter(township_name == name)
+  create_summary_table(township_data, added_feature)
+})
+```
+
+
+### Overall SHAP Summary
```

Comment:
These have interactive elements, and I think that is what hinders the iterative formatting

Reply:
**suggestion:** FWIW, @jeancochrane figured out a way to do this for leaflet maps that may also work in this case.